src/ucp/amo/amo.inl

```diff
@@ -43,80 +88,73 @@ ucs_status_t ucp_amo_check_send_status(ucp_request_t *req, ucs_status_t status);
         \
         req->send.lane = rkey->cache.amo_lane; \
         status = _function(ep->uct_eps[req->send.lane], \
-                           UCS_PP_TUPLE_BREAK _params, \
+                           value, *result, \
```

Comment:
why need to pass both result and *result?
also, does "cswap" progress have to be generic now? or maybe it can call a specific function?

Reply:
implemented as generic function
cswap requires 3 arguments: compare, value, result, UCP uses inplace argument: result is in-out argument